crates/executor/client/src/lib.rs

```diff
@@ -62,23 +58,28 @@ impl ClientExecutor {
         let cache_db = CacheDB::new(&witness_db);
 
         // Execute the block.
-        let spec = Arc::new(mainnet());
-        let current_block = profile!("recover senders", {
-            input.input.current_block.clone().try_into_recovered()
-        })?;
+        let spec = Arc::new(match chain_variant {
+            ChainVariant::Mainnet => mainnet(),
+            ChainVariant::Dev => dev(),
+        });
+        let current_block =
+            profile!("recover senders", { input.input.current_block.clone().try_into_recovered() })
```

Comment:
side note: `profile!` does not actually profile for us since we don't collect cycles in this way, so it's just a wasted println. I think we can delete them in a separate pr